packages/eas-cli/src/commands/build/resign.ts

```diff
@@ -0,0 +1,292 @@
+    const { limit, offset, nonInteractive } = flags;
+
+    const {
+      loggedIn: { actor, graphqlClient },
+      getDynamicProjectConfigAsync,
+      projectDir,
+      analytics,
+    } = await this.getContextAsync(BuildResign, {
+      nonInteractive: flags.nonInteractive,
+    });
+
+    const maybeBuild = flags.maybeBuildId
+      ? await this.maybeGetBuildAsync(graphqlClient, flags.maybeBuildId)
+      : undefined;
+    const platform =
+      (maybeBuild?.platform.toLowerCase() as Platform | undefined) ??
+      (await selectPlatformAsync(flags.platform));
+    if (platform === Platform.ANDROID) {
+      throw new Error('Re-signing archives is not supported on Android yet.');
+    }
+
+    await handleDeprecatedEasJsonAsync(projectDir, flags.nonInteractive);
+
+    await maybeWarnAboutEasOutagesAsync(graphqlClient, [StatuspageServiceName.EasBuild]);
+    const easJsonAccessor = new EasJsonAccessor(projectDir);
+    const easJsonCliConfig: EasJson['cli'] =
+      (await EasJsonUtils.getCliConfigAsync(easJsonAccessor)) ?? {};
+
+    const buildProfile = await EasJsonUtils.getBuildProfileAsync(
+      easJsonAccessor,
+      platform,
+      flags.profile ?? 'production'
+    );
+    const { exp, projectId } = await getDynamicProjectConfigAsync({ env: buildProfile.env });
+    const account = await getOwnerAccountForProjectIdAsync(graphqlClient, projectId);
+    const build = await this.ensureBuildSelectedAsync(
+      { graphqlClient, projectId, platform, nonInteractive, limit, offset },
+      maybeBuild
+    );
+    const credentialsCtx = new CredentialsContext({
+      projectInfo: { exp, projectId },
+      nonInteractive,
+      projectDir,
+      user: actor,
+      graphqlClient,
+      analytics,
+      env: buildProfile.env,
+      easJsonCliConfig,
+    });
+    if (buildProfile.credentialsSource !== CredentialsSource.LOCAL && !nonInteractive) {
+      await credentialsCtx.appStore.ensureAuthenticatedAsync();
+    }
+    const xcodeBuildContext = await resolveXcodeBuildContextAsync(
+      {
+        projectDir,
+        nonInteractive,
+        exp,
+      },
+      buildProfile
+    );
+    const targets = await resolveTargetsAsync({
+      projectDir,
+      exp,
+      xcodeBuildContext,
+      env: buildProfile.env,
+    });
+    const credentialsResult = await ensureIosCredentialsForBuildResignAsync(
+      credentialsCtx,
+      targets,
+      buildProfile
+    );
+    assert(build.artifacts?.applicationArchiveUrl, 'Missing application archive.');
+    const jobOverrides: IosJobOverridesInput = {
+      mode: BuildMode.Resign,
+      type: BuildWorkflow.Unknown,
+      resign: {
+        applicationArchiveSource: {
+          type: ProjectArchiveSourceType.Url,
+          url: build.artifacts?.applicationArchiveUrl,
+        },
+      },
+      secrets: prepareCredentialsToResign(credentialsResult.credentials),
+      builderEnvironment: { image: 'default' },
```

Comment:
Why? To always use the latest image?

Reply:
to make sure all the resigns happen in the same environment, we don't want to debug any issues because someone installed old version of fastlane.